cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -94,11 +104,15 @@ public void onConfigChange(List<InstanceConfig> configs, NotificationContext cha
 
   @Override
   @PreFetch(enabled = false)
-  public void onExternalViewChange(List<ExternalView> externalViewList, NotificationContext changeContext) {
+  public void onExternalViewChange(List<ExternalView> externalViewList,
+                                   NotificationContext changeContext) {
     generateShardConfig();
   }
 
   private void generateShardConfig() {
```

Comment:
This method should be synchronized.

Reply:
Are you worrying multiple threads will be working on `generateShardConfig` at the same time?

Reply:
yes.

Reply:
I will leave this method as it is and limit current pull request to be stats only. 
If future stats shows that we need synchronize this method, we can add synchronization then.